src/app/src/Map.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
+import ReactMapGL from 'react-map-gl';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import { connect } from 'react-redux';
+
+import onChangeViewport from './map.actions';
+import './Map.scss';
+
+class Map extends Component {
+    render() {
+        return (
+            <div id='map'>
+                <ReactMapGL
+                    height={'100%'}
+                    width={'100%'}
+                    {...this.props.viewport}
+                    onViewportChange={viewport =>
+                        this.props.updateViewport({ viewport })
+                    }
+                    mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_API_KEY}
+                />
+            </div>
+        );
+    }
+}
+
+function mapStateToProps(state) {
+    return state.map.viewport.toJS();
+}
+
+function mapDispatchToProps(dispatch) {
+    return {
+        updateViewport: viewport => dispatch(onChangeViewport(viewport)),
```

Comment:
Conceivably you could have this function do a bit more of the restructuring of the args --- essentially so that the `onViewportChange` line above would be

```
onViewportChange={this.props.updateViewport}
```

Reply:
Do you have an example?

Reply:
Sure -- so right now the invocation's set up to wrap the viewport in an object:

```
onViewportChange={viewport =>
                        this.props.updateViewport({ viewport })
                    }
```

...and then this function's set up to put that new object into the dispatched function.

I think you can adjust this line to be:

```
updateViewport: viewport => dispatch(onChangeViewport({ viewport })),
```

and then the invocation in the function would be able to do

```
viewport => this.props.updateViewport(viewport)
```

which reduces to 

```
this.props.updateViewport
```

Reply:
Ah, thanks! That's a nice, subtle way to clean-up the component. Pushed an update.